plugins/discourse-presence/assets/javascripts/discourse/components/composer-presence-display.gjs

```diff
@@ -0,0 +1,149 @@
+import Component from "@glimmer/component";
+import { cached, tracked } from "@glimmer/tracking";
+import { service } from "@ember/service";
+import { modifier } from "ember-modifier";
+import { gt } from "truth-helpers";
+import UserLink from "discourse/components/user-link";
+import avatar from "discourse/helpers/avatar";
+import i18n from "discourse-common/helpers/i18n";
+
+export default class ComposerPresenceDisplay extends Component {
+  @service presence;
+  @service composerPresenceManager;
+  @service currentUser;
+  @service siteSettings;
+
+  @tracked replyChannel;
+  @tracked whisperChannel;
+  @tracked editChannel;
+
+  setupReplyChannel = modifier(() => {
+    const topic = this.args.model.topic;
+    if (!topic || !this.isReply) {
+      return;
+    }
+
+    const replyChannel = this.presence.getChannel(
+      `/discourse-presence/reply/${topic.id}`
+    );
+    replyChannel.subscribe();
+    this.replyChannel = replyChannel;
+
+    return () => replyChannel.unsubscribe();
+  });
+
+  setupWhisperChannel = modifier(() => {
+    if (
+      !this.args.model.topic ||
+      !this.isReply ||
+      !this.currentUser.staff ||
+      !this.currentUser.whisperer
+    ) {
+      return;
+    }
+
+    const whisperChannel = this.presence.getChannel(
+      `/discourse-presence/whisper/${this.args.model.topic.id}`
+    );
+    whisperChannel.subscribe();
+    this.whisperChannel = whisperChannel;
+
+    return () => whisperChannel.unsubscribe();
+  });
+
+  setupEditChannel = modifier(() => {
+    if (!this.args.model.post || !this.isEdit) {
+      return;
+    }
+
+    const editChannel = this.presence.getChannel(
+      `/discourse-presence/edit/${this.args.model.post.id}`
+    );
+    editChannel.subscribe();
+    this.editChannel = editChannel;
+
+    return () => editChannel?.unsubscribe();
```

Comment:
Why does this one use the `?.` operator when the other two don't?

Reply:
right, it isn't needed anymore. removing!